qiita_db/analysis.py

```diff
@@ -958,7 +975,35 @@ def _build_biom_tables(self, grouped_samples, rename_dup_samples=False):
                     new_table.to_hdf5(
                         f, "Generated by Qiita, analysis id: %d, info: %s" % (
                             self._id, label))
-                biom_files.append((data_type, biom_fp))
+
+                # post_processing_cmds are a list of commands to run on the
+                # final BIOM. The order of operations is list-order. Each
+                # element of the list is a dictionary containing the Conda env
+                # to use, the script to run, and a dictionary of parameters.
+                if post_processing_cmds:
+                    print(type(post_processing_cmds))
+                    for cmd in post_processing_cmds:
+                        # for now, assume BIOM file is the last parameter
+                        # and assume biom_fp is overwritten
+                        params = ' '.join(["%s=%s" % (k, v) for k, v in
+                                          cmd['script_params'].items()])
+                        params = "%s %s" % (params, biom_fp)
+                        # if environment is successfully activated,
+                        # run script with parameters
+                        p = Popen(["source activate %s && python %s %s" % (
```

Comment:
This is given by the software itself and should not be hard-coded; for example, we have environments in conda and in virtualenv. In other words, you just need cmd['script_env'] and cmd['script_path'] separated by semicolon, nothing else.